### PR TITLE
Interop: Handle reveal and open in backend

### PR DESCRIPTION
**Pull request by latenssi**

### CODE REVIEW

File: e2e_test.go [hunk 1]
```diff
@@ -237,68 +257,23 @@ func TestE2E(t *testing.T) {
...
 		t.Fatal(err)
 	}
 
-	events = util.ParseTestEvents(e)
-	util.NewExpectedPackNFTEvent("Revealed").
-		AddField("id", randomPackID.String()).
-		AddField("salt", randomPack.Salt.String()).
-		AssertEqual(t, events[0])
+	if pack.State != common.PackStateRevealed {
+		t.Fatal("expected pack to revealed")
```
Comment (nanuuki) [inline e2e_test.go:271]:
```suggestion
		t.Fatal("expected pack to be revealed")
```

File: e2e_test.go [hunk 2]
```diff
@@ -315,26 +290,31 @@ func TestE2E(t *testing.T) {
...
 
-	// There are withdraw and deposit event for each nft being released from escrow
-	// So we only check the last event as "Opened"
-	l := len(randomPackNftIDsArr.ToGoValue().([]interface{})) * 2
-	events = util.ParseTestEvents(e)
-	util.NewExpectedPackNFTEvent("Opened").
-		AddField("id", randomPackID.String()).
-		AssertEqual(t, events[l])
+	if pack.State != common.PackStateOpened {
+		t.Fatal("expected pack to opened")
```
Comment (nanuuki) [inline e2e_test.go:304]:
```suggestion
		t.Fatal("expected pack to be opened")
```

File: service/flow_helpers/account.go [hunk 3]
```diff
@@ -34,10 +36,15 @@ func GetAccount(address flow.Address, privateKeyInHex string, keyIndexes []int)
...
+	randomIndex := rand.Intn(len(keyIndexes))
+
 	new := &Account{
-		Address:         address,
-		PrivateKeyInHex: privateKeyInHex,
-		KeyIndexes:      keyIndexes,
+		Address:           address,
+		PrivateKeyInHex:   privateKeyInHex,
+		KeyIndexes:        keyIndexes,
+		nextKeyIndexIndex: randomIndex,
```
Comment (nanuuki) [inline service/flow_helpers/account.go:47]: This _looks_ like a typo but probably isn't 👀

**Review by nanuuki — APPROVED** [review]
Nice! Nothing major ✔️

File: service/flow_helpers/account.go (hunk 3 above)
Comment (latenssi) [reply service/flow_helpers/account.go:47]: Yea... it is not a typo :)

File: e2e_test.go (hunk 2 above)
Comment (latenssi) [reply e2e_test.go:304]: Thanks!

File: e2e_test.go (hunk 1 above)
Comment (latenssi) [reply e2e_test.go:271]: Thanks!

**Pull request merged**